pkg/karmadactl/deinit/deinit.go

```diff
@@ -81,6 +82,7 @@ func NewCmdDeInit(parentCommand string) *cobra.Command {
 	flags.StringVar(&opts.Context, "context", "", "The name of the kubeconfig context to use")
 	flags.BoolVar(&opts.DryRun, "dry-run", false, "Run the command in dry-run mode, without making any server requests.")
 	flags.BoolVarP(&opts.Force, "force", "f", false, "Reset cluster without prompting for confirmation.")
+	flags.BoolVar(&opts.IgnoreNamespace, "ignore-namespace", false, "Reset cluster without delete namespace.")
```

Comment:
The flag name `ignore-namespace` might be ambiguous, since `deinit` is going to uninstall/remove `components` from a cluster. `ignore-namespace` can be misinterpreted as `if the component in the namespace, then ignore(cancel the process)`.  What do you think?

Reply:
This is indeed a question, what do you think of the flag name `no-delete-namespace`?

Reply:
Before talking about the flag name, I'd like to invite owners(@carlory @lonelyCZ ) to comment if this is a reasonable feature.

Reply:
--preserve-resources ?

Reply:
Mybe we can not delete namespace by default, and  delete namespace by specify options `--purge-namespace` ?

Reply:
kindly ping @lonelyCZ @my-git9, do we have an executive decision?

Reply:
>For security and more intuitive, I think we could let users determine if remove the namespace. Like

+1, istio does not delete `istio-system` namespace when it is uninstalled by istioctl

Reply:
> Mybe we can not delete namespace by default, and delete namespace by specify options --purge-namespace ?

I think it is ok.